homeassistant/components/homekit/thermostats.py

```diff
@@ -0,0 +1,137 @@
+            self.homekit_target_heat_cool = value
+            hass_value = HC_HOMEKIT_TO_HASS[value]
+
+            self._hass.services.call('climate', 'set_operation_mode',
+                                     {'entity_id': self._entity_id,
+                                      'operation_mode': hass_value})
+
+    def set_temperature_debounced(self, value):
+        _LOGGER.debug("%s: Call to set temperature to %.2f",
+                      self._entity_id, value)
+
+        def perform_call():
+            self.set_target_temperature(value)
+        # Cancel timer if one was running
+        if self._call_timer is not None:
+            try:
+                self._call_timer.cancel()
+            except AttributeError:
+                pass
+        self._call_timer = Timer(1, perform_call)
+        self._call_timer.start()
+
+    def set_target_temperature(self, value):
+        """Set target temperature to value if call came from HomeKit."""
+        if value != self.target_temperature:
+            _LOGGER.debug("%s: Set target temperature to %.2f",
+                          self._entity_id, value)
+            self.homekit_target_temperature = value
+            self._hass.services.call('climate', 'set_temperature',
+                                     {'entity_id': self._entity_id,
+                                      'temperature':
+                                          self.homekit_target_temperature})
+
+    def update_thermostat(self, entity_id=None,
+                          old_state=None, new_state=None):
+        """Update security state after state changed."""
+        if new_state is None:
+            return
+
+        current_temp = new_state.attributes[ATTR_CURRENT_TEMPERATURE]
+        if current_temp is not None:
+            self.current_temperature = float(current_temp)
+            self.char_current_temp.set_value(self.current_temperature)
+        target_temp = new_state.attributes[ATTR_TEMPERATURE]
+        if target_temp is not None:
+            self.target_temperature = float(target_temp)
+            self.char_target_temp.set_value(self.target_temperature)
+        operation_mode = new_state.attributes[ATTR_OPERATION_MODE]
+        if operation_mode is not None and operation_mode in HC_HASS_TO_HOMEKIT:
+            self.current_heat_cool = HC_HASS_TO_HOMEKIT[operation_mode]
+            self.char_current_heat_cool.set_value(self.current_heat_cool)
+        display_units = new_state.attributes['unit_of_measurement']
+        if display_units is not None and display_units in UNIT_HASS_TO_HOMEKIT:
+            self.display_units = UNIT_HASS_TO_HOMEKIT[display_units]
+            self.char_display_units.set_value(self.display_units)
+
+        if (self.homekit_target_heat_cool is None
+            and self.current_heat_cool is not None) \
+                or self.homekit_target_heat_cool == self.current_heat_cool:
+            self.char_target_heat_cool.set_value(self.current_heat_cool,
```

Comment:
What if `self.current_heat_cool = None`? Should result in an value error at the moment.